test/extended/util/cli.go

```diff
@@ -0,0 +1,221 @@
+func (c *CLI) KubeFramework() *Framework {
+	return c.kubeFramework
+}
+
+// SetNamespace overrides the namespace set by framework
+func (c *CLI) SetNamespace(ns string) {
+	if c.kubeFramework == nil {
+		FatalErr(fmt.Errorf("The E2E framework must be initialized"))
+	}
+	c.kubeFramework.Namespace = &kapi.Namespace{ObjectMeta: kapi.ObjectMeta{Name: ns}}
+}
+
+func (c *CLI) beforeEach() {
+	if len(c.Namespace()) > 0 {
+		Logf("Adding project role bindings to %q namespace", c.Namespace())
+		c.SetupRoleBindings()
+	} else {
+		Failf("Framework does not have the namespace set")
+	}
+}
+
+// SetupRoleBindings setups a project role binding for the current namespace
+func (c *CLI) SetupRoleBindings() *CLI {
+	for _, binding := range bootstrappolicy.GetBootstrapServiceAccountProjectRoleBindings(c.Namespace()) {
+		addRole := &policy.RoleModificationOptions{
+			RoleName:            binding.RoleRef.Name,
+			RoleNamespace:       binding.RoleRef.Namespace,
+			RoleBindingAccessor: policy.NewLocalRoleBindingAccessor(c.Namespace(), c.AdminRESTClient()),
+			Users:               binding.Users.List(),
+			Groups:              binding.Groups.List(),
+		}
+		if err := addRole.AddRole(); err != nil {
+			Failf("Unable to add role binding %+v to namespace %q: %v", addRole, c.Namespace(), err)
+		}
+	}
+	return c
+}
+
+// Verbose turns on verbose messages for current command
+func (c *CLI) Verbose() *CLI {
+	c.verbose = true
+	return c
+}
+
+// AdminRESTClient return the current project namespace REST client
+func (c *CLI) AdminRESTClient() *client.Client {
+	client, err := testutil.GetClusterAdminClient(c.adminConfigPath)
+	if err != nil {
+		FatalErr(err)
+		return nil
+	}
+	return client
+}
+
+// AdminKubeRESTClient returns the current project namespace Kubernetes client
+func (c *CLI) AdminKubeRESTClient() *kclient.Client {
+	if c.kubeFramework.Client != nil {
+		return c.kubeFramework.Client
+	}
+	client, err := testutil.GetClusterAdminKubeClient(c.adminConfigPath)
+	if err != nil {
+		FatalErr(err)
+		return nil
+	}
+	return client
+}
+
+// Namespace returns the name of the namespace used in the current test case
+func (c *CLI) Namespace() string {
+	if c.kubeFramework == nil {
+		return ""
+	}
+	if c.kubeFramework.Namespace == nil {
+		return ""
+	}
+	return c.kubeFramework.Namespace.Name
+}
+
+// SetOutput sets the default output for the command
+func (c *CLI) SetOutput(out io.Writer) *CLI {
+	c.stdout = out
+	for _, subCmd := range c.cmd.Commands() {
+		subCmd.SetOutput(c.stdout)
+	}
+	c.cmd.SetOutput(c.stdout)
+	return c
+}
+
+// Run executes given OpenShift CLI command verb (iow. "oc <verb>").
+// This function also override the default 'stdout' to redirect all output
+// to a buffer and prepare the global flags such as namespace and config path.
+func (c *CLI) Run(verb string) *CLI {
+	out := new(bytes.Buffer)
+	if len(c.Namespace()) == 0 {
+		FatalErr(fmt.Errorf("You must set the namespace before running a command."))
+	}
+	nc := &CLI{
```

Comment:
Why are you creating and return a new cli struct here? Isn't every method supposed to do it's stuff on the struct and return it instead of returning a new struct?